test/e2e/network_segmentation.go

```diff
@@ -508,6 +509,84 @@ var _ = Describe("Network Segmentation", func() {
 		expectedMessage := fmt.Sprintf("primary network already exist in namespace %q: %q", f.Namespace.Name, primaryNadName)
 		Expect(actualConditions[0].Message).To(Equal(expectedMessage))
 	})
+
+	Context("pod2Egress on a user defined primary network", func() {
+		const (
+			externalContainerName = "ovn-k-egress-test-helper"
+		)
+		var externalIpv4 string
+		BeforeEach(func() {
+			externalIpv4, _ = createClusterExternalContainer(
+				externalContainerName,
+				"registry.k8s.io/e2e-test-images/agnhost:2.45",
+				runExternalContainerCmd(),
+				httpServerContainerCmd(port),
+			)
+
+			DeferCleanup(func() {
+				deleteClusterExternalContainer(externalContainerName)
+			})
+		})
+
+		DescribeTable(
+			"can be accessed to from the pods running in the Kubernetes cluster",
+			func(netConfigParams networkAttachmentConfigParams, clientPodConfig podConfiguration) {
+				netConfig := newNetworkAttachmentConfig(netConfigParams)
+
+				netConfig.namespace = f.Namespace.Name
+				clientPodConfig.namespace = f.Namespace.Name
+
+				By("creating the attachment configuration")
+				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
```

Comment:
can we use the UDN CRD instead now?

Reply:
yeah this e2e was written long before we had the CRDs... I don't have time to work on this...as part of this PR
https://redhat-internal.slack.com/archives/C077NGF5HE1/p1723116215316809?thread_ts=1723104646.713079&cid=C077NGF5HE1
we can do it as FUP if its ok with you...
@npinaeva can you help convert this test to use CRD when this PR merges?

Reply:
https://github.com/ovn-org/ovn-kubernetes/issues/4606

Reply:
will be solved via FUP